x-pack/plugin/otel-data/src/yamlRestTest/resources/rest-api-spec/test/20_metic_tests.yml

```diff
@@ -0,0 +1,48 @@
+---
+setup:
+  - do:
+      cluster.health:
+        wait_for_events: languid
+
+---
+"Test push service overview metric":
+  - do:
+      indices.put_index_template:
+        name: test@template
+        body:
+          priority: 500
+          index_patterns: [ "metrics-*.otel-*" ]
+          composed_of:
+            - metrics@tsdb-settings
+            - otel@mappings
+            - metrics@mappings
+            - metrics-otel@mappings
+            - semconv-resource-to-ecs@mappings
+            - ecs-tsdb@mappings
+          template:
+            settings:
+              index:
+                routing_path: [unit, attributes.*, resource.attributes.*]
+                mode: time_series
+                time_series:
+                  start_time: 2024-07-01T13:03:08.138Z
```

Comment:
The challenge here is that we need to set `time_series.start_time`, or use dynamic `@timestamp` to avoid rejecting too old `@timestamp` by TSDB.

I tried:
- Updating the existing `metrics-otel@template`, which would be the ideal solution. If I just use `indices.put_index_template` with the existing `metrics-otel@template` name, it gets overwritten, so does not seem to work.
- Tried to push a `metrics-otel@custom` mapping - I was not able to figure out how to do that with this `yml` syntax. Haven't found any sample in the repo and it got very painful to test it over and over. This would be still fine in my opinion, so if someone knows how to do so, it'd be great.
- Alternatively we could look into how to use dynamic `@timestamps` would be easy in Java - or in any other language - but so far I have not found a way to do that in `yml`.

With that, the remaining idea was to re-create the whole mapping with setting `time_series.start_time`. But overall I think this is not very good.

Reply:
> Tried to push a metrics-otel@custom mapping - I was not able to figure out how to do that with this yml syntax. 
When you search for `cluster.put_component_template`, you can find a bunch of examples.

It should be something similar to 

```yml
  - do:
      cluster.put_component_template:
        name: metrics-otel@custom
        body:
          template:
            settings:
              index:
                time_series:
                  start_time: 2024-07-01T13:03:08.138Z
```

If setting the start_time doesn't work here, I have another idea which involves setting the `@timestamp` in an ingest pipeline.

```yml
  - do:
      ingest.put_pipeline:
        id: "default_timestamp"
        body:  >
          {
            "description": "_description",
            "processors": [
              {
                "set": {
                  "description": "If '@timestamp' is missing, set it with the ingest timestamp",
                  "field": "@timestamp",
                  "override": false,
                  "copy_from": "_ingest.timestamp"
                }
              }
            ]
          }
  - match: { acknowledged: true }

  - do:
      cluster.put_component_template:
        name: metrics-otel@custom
        body:
          template:
            settings:
              index:
                default_pipeline: default_timestamp
```

Reply:
Nice, thanks! Pushing a custom `metrics-otel@custom` worked - I think I was searching for `indices.put_component_template` and not `cluster.put_component_template`. Anyways, this works.